ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -435,17 +436,61 @@ def create_discrete(discrete_branches: Tuple[int]) -> "ActionSpec":
         return ActionSpec(0, discrete_branches)
 
 
-class BehaviorSpec(NamedTuple):
+class DimensionProperty(IntFlag):
     """
-    A NamedTuple containing information about the observation and action
-    spaces for a group of Agents under the same behavior.
+    No properties specified.
+    """
+
+    UNSPECIFIED = 0
+    """
+    No Property of the observation in that dimension. Observation can be processed with
+    Fully connected networks.
+    """
+    NONE = 1
+    """
+    Means it is suitable to do a convolution in this dimension.
+    """
+    TRANSLATIONAL_EQUIVARIANCE = 2
+    """
+    Means that there can be a variable number of observations in this dimension.
+    The observations are unordered.
+    """
+    VARIABLE_SIZE = 4
+
+
+class ObservationSpec(NamedTuple):
```

Comment:
I'm wondering if rather than having the ObservationSpec hold multiple `List[item]`, we have the `BehaviorSpec` hold a `List[ObservationSpec]` and have each ObservationSpec hold the information for one obs. That way, when we go through and e.g. create the input processors based on the DimensionProperty and type, we can do something like:

```
for obs_spec in observation_specs:
    _shape = obs_spec.shape
    _properties = obs_spec.dimension_properties
     _type = obs_spec.type # TBD
```

Rather than 
```
for _shape, _properties, _type in zip(observation_spec.shapes, observation_spec.dimension_properties, observation_spec.types):
```

Reply:
> we have the BehaviorSpec hold a List[ObservationSpec] and have each ObservationSpec hold the information for one obs. 

I think I like this better too.